sdk/opa_test.go

```diff
@@ -805,38 +1171,36 @@ main = 7
 		}
 	}`, server.URL())
 
-	testLogger := loggingtest.New()
 	opa, err := sdk.New(ctx, sdk.Options{
-		Config:        strings.NewReader(config),
-		ConsoleLogger: testLogger,
+		Config: strings.NewReader(config),
 	})
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	defer opa.Stop(ctx)
 
-	// Execute two queries.
-	if _, err := opa.Decision(ctx, sdk.DecisionOptions{}); err != nil {
+	// Execute two queries with metrics
+	m1 := metrics.New()
+	if _, err := opa.Decision(ctx, sdk.DecisionOptions{
+		Metrics: m1,
+	}); err != nil {
 		t.Fatal(err)
 	}
 
-	if _, err := opa.Decision(ctx, sdk.DecisionOptions{}); err != nil {
+	m2 := metrics.New()
+	if _, err := opa.Decision(ctx, sdk.DecisionOptions{
+		Metrics: m2,
+	}); err != nil {
 		t.Fatal(err)
 	}
 
-	// Expect two log entries, one with timers for query preparation and the other without.
-	entries := testLogger.Entries()
-
-	if len(entries) != 2 {
-		t.Fatal("expected two log entries but got:", entries)
+	// Expect only the metrics from the first query to contain preparation metrics
+	if _, ok := m1.All()["timer_rego_query_parse_ns"]; !ok {
```

Comment:
Since this test was looking for metrics in the console logger, I thought that it might make more sense to use the metrics instead.

Reply:
Yup, nice cleanup!